Two collab sdk dev fixes

keith-packard opened the pull request:
* Align iterable objects to 8 bytes when using 64 bit timeouts
 * Increase test stack size when using c++ exceptions

nashif commented:
@keith-packard  had to cherry-pick one of the patches for now, this same branch is being used for the 12.2 gcc and is failing with the `_static_thread_data` patch. Need another branch for gcc-14 changes. It is getting messy and with too many things happening at the same time

keith-packard commented:
the _static_thread_data patch is needed for sparc as that assumes global variables are 8-byte aligned (!). We could disable that in the compiler...

keith-packard committed:
modules/cmsis-dsp: Don't use Zephyr stdint.h

Zephyr replaces the toolchain version of stdint.h to define uint32_t and
int32_t as int rather than long. This breaks the ARM MVE intrinics which
require uint32_t to be defined as unsigned long.

Provide 'ZEPHYR_USE_TOOLCHAIN_STDINT' which causes zephyr_stdint.h to be
skipped, ensuring that the normal stdint.h types are used instead.

Signed-off-by: Keith Packard <keithp@keithp.com>
